### DISCUSSION TITLE
synchronizing mbed-ce fork and PR's

### DISCUSSION THREAD

**Discussion by JojoS62** [mbed-ce/mbed-os#90] [General]
How should PRs be handled?
E.g., my current project is a controller for my house heating, using a STM32F769 Disco board. It works with lvgl and network, but there is a problem that ethernet stops after a few minutes. I found a similar issue in mbed-os, and a workaround with mapping the ethernet descriptors and buffers in SRAM2 works. But this requires two modifications: use a section for the buffer declaration and change the linker script.
PR for stm32xx_emap.cpp: 
- to mbed-os: needs running a lot of tests, who will care now?
- to mbed-ce: maybe easier, but then how can the forks be in sync? The change belongs to the base mbed-os, is a rebase working without a lot of merge conflicts?

@multiplemonomials how did you handle this so far? I see a lot of commits ahead, this is really great work, but seems difficult to maintain.

**Comment by multiplemonomials** [comment]
I mean, this is kind a difficult issue.  In my experience so far, Mbed OS has not been very receptive to significant changes without contributors jumping through a huge number of hoops.  So, I don't plan to try and merge most of the Mbed CE changes back to Mbed OS.  Maybe I'll try to get some bugfixes here and there though, like my STM32 I2C rewrite.

**Comment by JojoS62** [comment]
yes, but I mean more the reverse way, rebasing on newer mbed-os. There could be bug fixes, but I guess there are no big changes. So important changes will be imported by cherry picking?

**Reply by multiplemonomials** [reply]
@jay-sridharan is working on this, I believe his plan is to rebase mbed-ce on top of mbed-os every few months.

**Reply by jay-sridharan** [reply]
Yes, from what it looks like, the upstream mbed-os has just small fixes here and there which we can easily rebase on. If mbed-os begins active feature development again or any kind of large scale refactoring, we will have to assess the situation then. Until then, rebasing keeps us in a position to be flexible.

**Reply by JojoS62** [reply]
thanks, I have used mbed-ce now for a few weeks and it is great, the workflow and the good integration into VSC with cmake-tools makes it very handy. And I'm sure that it will not be a big problem to keep it up to date.
